feat(docs): SCIM Azure setup documentation (#3866)
getsentry/sentry-docs: src/docs/product/accounts/sso/azure-sso.mdx

1. In the search bar, search for "Groups" then navigate to it. From there, create a new group, add an owner and members to the group. Set "Group type" to Office 365. For more details about group creation, see the [Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

> 1. Return to the **Overview** page. In the search bar, enter `Enterprise Applications` and navigate to it. Click "+ New application". Search for `Sentry` to create the application for Sentry.

Reviewer: ```suggestion
1. Return to the **Overview** page. In the search bar, enter `Enterprise Applications`, then navigate to it. Click "+ New application". Search for `Sentry` to create the application.
```

    - First Name: `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname`

    - Last Name: `http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname`

      For more details about mappings for custom configs, see [The Role of Claims](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/technical-reference/the-role-of-claims).

   ![Map Identity with provider attributes](azure-map-identity-provider-attributes.png)

1. Sentry will attempt to authenticate and link your account with Azure. After successful authentication, you'll be redirected to Sentry's SSO configuration page, where you can take the following actions:

   - You can share the "Login URL" value, which will be used for SP-initiated SSO, with the users in your organization.

   - Scroll down to the bottom and ensure that "Require SSO" is checked if you want to enforce logging in with Okta.

   - Set a "Default Role" for new SSO users. Selecting "Member" should cover most use-cases.

   - If you made changes, click "Save Settings" to complete your setup.

## SCIM Integration

> <Note>

Reviewer: ```suggestion

<Note>
```

This feature is available only if you're in the Early Adopter program. Features available to Early Adopters are still in-progress and may have bugs. We recognize the irony.

If you’re interested in being an Early Adopter, you can turn your organization’s Early Adopter status on/off in General Settings. This will affect all users in your organization and can be turned back off just as easily.

> </Note>

Reviewer: ```suggestion
</Note>

```

Sentry users can manage provisioning using Azure with SCIM. You'll need to have Azure SSO set up and configured for your organization already. Sentry supports User and Group provisioning with Azure.

> ### Sentry Configuration

Author: It seems like this could use a bit of spacing from the above paragraph, but not sure how to do so within the markdown here.

Reviewer: yes, this does look a bit squashed. I'm not sure why. Try inserting spaces above the Note to see it renders (I'll add those lines as suggestions).

1. Sign in to [sentry.io](https://sentry.io). Select **Settings > Auth**
   ![Sentry Okta SCIM Auth Page](scim-settings-auth.png)
1. Under General Settings select "Enable SCIM", then "Save Settings"
   ![Sentry Enable SCIM Instruction](scim-enable-sentry.png)
Sentry will display "SCIM Information" that contains your Auth Token and SCIM Base URL.
   ![SCIM Credentials Fields](scim-sentry-fields.png)

### Azure Configuration

1. Log in to the [Azure portal](https://portal.azure.com/).

1. In your Azure portal, navigate to **Enterprise applications** in the sidebar.

1. Select the existing Sentry application.

> 1. Under the "Manage" menu, select "Provisioning".

Reviewer: ```suggestion
1. Select "Provisioning" from the "Manage" menu. 
```

>    ![Azure Provisioning Selection](azure-scim-provisioning.png)

Author: is this screenshot too big? is the best solution just to edit it down on my computer? not sure if the size is always expanded to a certain width or something.

Reviewer: Did you follow our guidelines for screenshots? I will send you a link to them :)

1. Set "Provisioning Mode" to "Automatic".

> 1. Take the values from your Sentry SCIM configuration and enter your SCIM URL in the "Tenant URL" field and your SCIM Auth token in the "Secret Token" field.

Reviewer: ```suggestion
1. Use the values from your Sentry SCIM configuration to enter your SCIM URL in the "Tenant URL" field and your SCIM Auth token in the "Secret Token" field.
```

1. Select "Test Connection" and wait for the message that confirms that the credentials are authorized to enable provisioning.
   ![Azure Test SCIM Connection](azure-scim-test-connection.png)

> 1. It's recommended that you select the "Send an email notification when a failure occurs" option.

Reviewer: This isn't really a step, so no need to number it.
```suggestion
We recommend that you select the option to "Send an email notification when a failure occurs".
```
